src/filters/sse.rs

```diff
@@ -40,24 +42,27 @@ use std::error::Error as StdError;
 use std::fmt::{self, Display, Formatter, Write};
 use std::str::FromStr;
 use std::time::Duration;
+use std::task::{Context, Poll};
+use std::pin::Pin;
+use std::future::Future;
 
-use futures::{Async, Future, Poll, Stream};
+use futures::{future, Stream, TryStream, TryStreamExt};
 use http::header::{HeaderValue, CACHE_CONTROL, CONTENT_TYPE};
 use hyper::Body;
 use serde::Serialize;
 use serde_json;
-use tokio::{clock::now, timer::Delay};
+use tokio::{clock::now, timer::{self, Delay}};
 
 use self::sealed::{
     BoxedServerSentEvent, EitherServerSentEvent, SseError, SseField, SseFormat, SseWrapper,
 };
 use super::header;
-use filter::One;
-use reply::Response;
-use {Filter, Rejection, Reply};
+use crate::filter::One;
+use crate::reply::Response;
+use crate::{Filter, Rejection, Reply};
 
 /// Server-sent event message
-pub trait ServerSentEvent: SseFormat + Sized + Send + 'static {
+pub trait ServerSentEvent: SseFormat + Sized + Send + Sync + 'static {
```

Comment:
Is there a reason for the `Sync` bound? Pre `std::Future` SSE replys did not have to be `Sync`.